timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TimelockNamespaces.java

```diff
@@ -108,6 +117,12 @@ public Set<Client> getActiveClients() {
         return services.keySet().stream().map(Client::of).collect(toSet());
     }
 
+    public Set<Client> getActiveClientsWithExpiry() {
+        Instant expiryTime = Instant.now().minus(ACTIVE_CLIENT_DURATION);
```

Comment:
A slightly smarter thing we could do, without going too crazy, is just track the last time we did this computation and just return the set as is if it was in the last hour or similar without recalculating the active entries.

This would limit damage to timelock if something happened to spam this endpoint over and over again.

(You'd need to set the expiry time to something like 4 hours 45 mins to preserve the required API guarantee!)

Reply:
(and if you do decide to do this, I'd just wrap it up in a nice abstraction to top it off)